libibumad/umad.h

```diff
@@ -54,6 +54,21 @@ typedef __be16 __attribute__((deprecated)) be16_t;
 typedef __be32 __attribute__((deprecated)) be32_t;
 typedef __be64 __attribute__((deprecated)) be64_t;
 
+typedef union _ib_gid
+{
+	uint8_t	raw[16];
+
+	struct {
+		__be64	prefix;
+		__be64	interface_id;
+	} unicast;
+
+	struct {
+		uint8_t	header[2];
+		uint8_t	raw_group_id[14];
+	} multicast;
+} ib_gid_t;
+
```

Comment:
The overall size may not have changed but I don't think that's sufficient. All the structs inside the data details union need to be properly located and I'm not sure they are without the packing. As Jason pointed out, some are not naturally aligned.